ApplicationNameBeanPostProcessor cannot access private properties

### Steps to reproduce

* camunda-bpm-taskpool version: 4.0.0
* Camunda BPM version: 7.20.1
* JDK version: 17.0.9

I have an application, which uses `io.axoniq.console:console-framework-client-spring-boot-starter` alongside Polyflow. The axon library contains a ConfigurationProperties class, which has a `private applicationName` property.

When I start the SpringBoot app, ApplicationNameBeanPostProcessor tries to access the `applicationName` property, but fails:

```
Caused by: kotlin.reflect.full.IllegalCallableAccessException: java.lang.IllegalAccessException: class kotlin.reflect.jvm.internal.calls.CallerImpl$FieldGetter cannot access a member of class io.axoniq.console.framework.starter.AxoniqConsoleSpringProperties with modifiers "private"
  at kotlin.reflect.jvm.internal.KCallableImpl.call(KCallableImpl.kt:280)
  at kotlin.reflect.jvm.internal.KProperty1Impl.get(KProperty1Impl.kt:35)
  at io.holunda.polyflow.spring.ApplicationNameBeanPostProcessor.postProcessAfterInitialization(ApplicationNameBeanPostProcessor.kt:29)
  at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization(AbstractAutowireCapableBeanFactory.java:434)
  at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1773)
  at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:598)
  ... 116 common frames omitted
Caused by: java.lang.IllegalAccessException: class kotlin.reflect.jvm.internal.calls.CallerImpl$FieldGetter cannot access a member of class io.axoniq.console.framework.starter.AxoniqConsoleSpringProperties with modifiers "private"
  at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:392)
  at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:674)
  at java.base/java.lang.reflect.Field.checkAccess(Field.java:1102)
  at java.base/java.lang.reflect.Field.get(Field.java:423)
  at kotlin.reflect.jvm.internal.calls.CallerImpl$FieldGetter.call(CallerImpl.kt:176)
  at kotlin.reflect.jvm.internal.KCallableImpl.call(KCallableImpl.kt:108)
  ... 121 common frames omitted
```

--- a/bom/parent/pom.xml
+++ b/bom/parent/pom.xml
@@ -18,10 +18,10 @@
   <url>https://github.com/holunda-io/camunda-bpm-taskpool/</url>
 
   <properties>
-    <springboot.version>3.1.4</springboot.version>
+    <springboot.version>3.1.5</springboot.version>
     <camunda-commons-typed-values.version>7.20.0</camunda-commons-typed-values.version>
 
-    <axon-bom.version>4.8.3</axon-bom.version>
+    <axon-bom.version>4.9.0</axon-bom.version>
     <axon-kotlin.version>4.9.0</axon-kotlin.version>
     <axon-gateway-extension.version>2.0.0</axon-gateway-extension.version>
 

--- a/core/bus-jackson/src/test/kotlin/io/holunda/polyflow/view/query/data/DataEntryQueriesDeserializationTest.kt
+++ b/core/bus-jackson/src/test/kotlin/io/holunda/polyflow/view/query/data/DataEntryQueriesDeserializationTest.kt
@@ -33,12 +33,12 @@ internal class DataEntryQueriesDeserializationTest {
         DataEntriesForUserQuery(
           user = User(
             username = "kermit", groups = setOf("muppets")
-          ), page = 1, size = 50, sort = "+name", filters = listOf("data.name=test")
+          ), page = 1, size = 50, sort = listOf("+name"), filters = listOf("data.name=test")
         )
       ), Arguments.of(
         DataEntriesForDataEntryTypeQuery::class.java,
         DataEntriesForDataEntryTypeQuery(
-          entryType = "domain.type", page = 1, size = 50, sort = "+name"
+          entryType = "domain.type", page = 1, size = 50, sort = listOf("+name")
         )
       )
     )

--- a/core/datapool/datapool-core/src/test/kotlin/io/holunda/polyflow/datapool/core/itest/TestApplication.kt
+++ b/core/datapool/datapool-core/src/test/kotlin/io/holunda/polyflow/datapool/core/itest/TestApplication.kt
@@ -3,6 +3,7 @@ package io.holunda.polyflow.datapool.core.itest
 import com.thoughtworks.xstream.XStream
 import com.thoughtworks.xstream.security.AnyTypePermission
 import io.holunda.polyflow.datapool.core.EnablePolyflowDataPool
+import org.axonframework.eventsourcing.eventstore.inmemory.InMemoryEventStorageEngine
 import org.axonframework.serialization.Serializer
 import org.axonframework.serialization.xml.XStreamSerializer
 import org.springframework.beans.factory.annotation.Qualifier
@@ -16,4 +17,6 @@ class TestApplication {
   @Qualifier("eventSerializer")
   fun myEventSerializerForProcess(): Serializer = XStreamSerializer.builder().xStream(XStream().apply { addPermission(AnyTypePermission.ANY) }).build()
 
+  @Bean
+  fun inMemoryStorageEngine() = InMemoryEventStorageEngine()
 }

--- a/core/spring-utils/src/main/kotlin/io/holunda/polyflow/spring/ApplicationNameBeanPostProcessor.kt
+++ b/core/spring-utils/src/main/kotlin/io/holunda/polyflow/spring/ApplicationNameBeanPostProcessor.kt
@@ -8,6 +8,10 @@ import org.springframework.core.annotation.AnnotationUtils
 import org.springframework.core.env.get
 import kotlin.reflect.KMutableProperty1
 import kotlin.reflect.full.memberProperties
+import kotlin.reflect.jvm.isAccessible
+import kotlin.reflect.jvm.javaField
+import kotlin.reflect.jvm.javaGetter
+import kotlin.reflect.jvm.javaSetter
 
 /**
  * A bean post processor that inspects all [ConfigurationProperties] beans, looks for a mutable property of type `String` called `applicationName` and replaces it with the value
@@ -17,15 +21,21 @@ class ApplicationNameBeanPostProcessor(private val applicationContext: Applicati
 
   companion object {
     const val UNSET_APPLICATION_NAME = "unset-application-name"
+    val CANDIDATES = arrayOf(
+      "io.holunda.polyflow.taskpool.collector.CamundaTaskpoolCollectorProperties",
+      "io.holunda.polyflow.client.camunda.CamundaEngineClientProperties",
+      "io.holunda.polyflow.datapool.DataEntrySenderProperties",
+    )
   }
 
   private val applicationName: String by lazy { applicationContext.environment["spring.application.name"] ?: UNSET_APPLICATION_NAME }
   override fun postProcessAfterInitialization(bean: Any, beanName: String): Any {
-    if (AnnotationUtils.findAnnotation(bean::class.java, ConfigurationProperties::class.java) != null) {
+    if (CANDIDATES.contains(bean::class.java.name)
+      && AnnotationUtils.findAnnotation(bean::class.java, ConfigurationProperties::class.java) != null) {
       @Suppress("UNCHECKED_CAST")
       val applicationNameProperty =
         bean::class.memberProperties.find { it.name == "applicationName" && it is KMutableProperty1<out Any, *> && it.returnType.classifier == String::class } as KMutableProperty1<Any, String>?
-      if (applicationNameProperty != null) {
+      if ( applicationNameProperty != null) {
         if (applicationNameProperty.get(bean) == UNSET_APPLICATION_NAME) {
           applicationNameProperty.set(bean, applicationName)
         }

--- a/core/spring-utils/src/test/kotlin/io/holunda/polyflow/datapool/DataEntrySenderProperties.kt
+++ b/core/spring-utils/src/test/kotlin/io/holunda/polyflow/datapool/DataEntrySenderProperties.kt
@@ -1,0 +1,7 @@
+package io.holunda.polyflow.datapool
+
+import io.holunda.polyflow.spring.ApplicationNameBeanPostProcessor
+import org.springframework.boot.context.properties.ConfigurationProperties
+
+@ConfigurationProperties(prefix = "polyflow.test")
+data class DataEntrySenderProperties(var applicationName: String = ApplicationNameBeanPostProcessor.UNSET_APPLICATION_NAME)

--- a/core/spring-utils/src/test/kotlin/io/holunda/polyflow/spring/ApplicationNameBeanPostProcessorTest.kt
+++ b/core/spring-utils/src/test/kotlin/io/holunda/polyflow/spring/ApplicationNameBeanPostProcessorTest.kt
@@ -1,5 +1,6 @@
 package io.holunda.polyflow.spring
 
+import io.holunda.polyflow.datapool.DataEntrySenderProperties
 import io.holunda.polyflow.spring.ApplicationNameBeanPostProcessor.Companion.UNSET_APPLICATION_NAME
 import org.assertj.core.api.Assertions.assertThat
 import org.junit.jupiter.api.Test
@@ -20,10 +21,19 @@ internal class ApplicationNameBeanPostProcessorTest {
       .withPropertyValues(
         "spring.application.name=my-test-application"
       ).run {
-        val testProperties = it.getBean(TestProperties::class.java)
+        val privateTestProperties = it.getBean(PrivateTestProperties::class.java)
+        assertThat(privateTestProperties).isNotNull
+        assertThat(privateTestProperties.getApplicationNamePropertyValue()).isEqualTo("unset-application-name")
+      }
+    contextRunner
+      .withPropertyValues(
+        "spring.application.name=my-test-application",
+      ).run {
+        val testProperties = it.getBean(DataEntrySenderProperties::class.java)
         assertThat(testProperties).isNotNull
         assertThat(testProperties.applicationName).isEqualTo("my-test-application")
       }
+
   }
 
   @Test
@@ -31,9 +41,19 @@ internal class ApplicationNameBeanPostProcessorTest {
     contextRunner
       .withPropertyValues(
         "spring.application.name=my-test-application",
+        "polyflow.test.private.application-name=my-polyflow-test-application"
+      ).run {
+        val privateTestProperties = it.getBean(PrivateTestProperties::class.java)
+        assertThat(privateTestProperties).isNotNull
+        assertThat(privateTestProperties.getApplicationNamePropertyValue()).isEqualTo("my-polyflow-test-application")
+      }
+
+    contextRunner
+      .withPropertyValues(
+        "spring.application.name=my-test-application",
         "polyflow.test.application-name=my-polyflow-test-application"
       ).run {
-        val testProperties = it.getBean(TestProperties::class.java)
+        val testProperties = it.getBean(DataEntrySenderProperties::class.java)
         assertThat(testProperties).isNotNull
         assertThat(testProperties.applicationName).isEqualTo("my-polyflow-test-application")
       }
@@ -43,7 +63,13 @@ internal class ApplicationNameBeanPostProcessorTest {
   fun `leaves application name at default if spring-application-name is not set`() {
     contextRunner
       .run {
-        val testProperties = it.getBean(TestProperties::class.java)
+        val privateTestProperties = it.getBean(PrivateTestProperties::class.java)
+        assertThat(privateTestProperties).isNotNull
+        assertThat(privateTestProperties.getApplicationNamePropertyValue()).isEqualTo(UNSET_APPLICATION_NAME)
+      }
+    contextRunner
+      .run {
+        val testProperties = it.getBean(DataEntrySenderProperties::class.java)
         assertThat(testProperties).isNotNull
         assertThat(testProperties.applicationName).isEqualTo(UNSET_APPLICATION_NAME)
       }
@@ -51,9 +77,11 @@ internal class ApplicationNameBeanPostProcessorTest {
 }
 
 @Configuration
-@EnableConfigurationProperties(TestProperties::class)
+@EnableConfigurationProperties(value = [PrivateTestProperties::class, DataEntrySenderProperties::class])
 @Import(ApplicationNameBeanPostProcessor::class)
 class TestConfig
 
-@ConfigurationProperties(prefix = "polyflow.test")
-data class TestProperties(var applicationName: String = UNSET_APPLICATION_NAME)
+@ConfigurationProperties(prefix = "polyflow.test.private")
+data class PrivateTestProperties(private var applicationName: String = UNSET_APPLICATION_NAME) {
+  fun getApplicationNamePropertyValue() = this.applicationName
+}

--- a/core/taskpool/taskpool-core/src/test/kotlin/io/holunda/polyflow/taskpool/core/TestApplication.kt
+++ b/core/taskpool/taskpool-core/src/test/kotlin/io/holunda/polyflow/taskpool/core/TestApplication.kt
@@ -2,6 +2,7 @@ package io.holunda.polyflow.taskpool.core
 
 import com.thoughtworks.xstream.XStream
 import com.thoughtworks.xstream.security.AnyTypePermission
+import org.axonframework.eventsourcing.eventstore.inmemory.InMemoryEventStorageEngine
 import org.axonframework.serialization.Serializer
 import org.axonframework.serialization.xml.XStreamSerializer
 import org.springframework.beans.factory.annotation.Qualifier
@@ -15,4 +16,6 @@ class TestApplication {
   @Qualifier("eventSerializer")
   fun myEventSerializerForProcess(): Serializer = XStreamSerializer.builder().xStream(XStream().apply { addPermission(AnyTypePermission.ANY) }).build()
 
+  @Bean
+  fun inMemoryEventStoreEngine() = InMemoryEventStorageEngine()
 }

--- a/pom.xml
+++ b/pom.xml
@@ -88,7 +88,7 @@
             <configuration>
               <rules>
                 <requireMavenVersion>
-                  <version>3.6.0</version>
+                  <version>3.9.0</version>
                 </requireMavenVersion>
                 <requireJavaVersion>
                   <version>17</version>

